### PR TITLE
renovate: group all dependencies into a single PR

### PR DESCRIPTION
**Pull request by dprince**

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -27,8 +28,7 @@
       "excludePackagePatterns": [
         "^github.com/openstack-k8s-operators"
```
Comment (stuggi) [inline renovate.json:29]: not sure I understand this excludePackagePatterns and the above matchPackageNames. Doesn't the above include it, but here we exclude it?

Comment (dprince) [reply renovate.json]: I've reworked this now to be explicit with the packages we are disabling.

File: renovate.json [hunk 2]
```diff
@@ -2,33 +2,32 @@
...
-      ],
-      "extends": ["schedule:monthly"],
-      "groupName": "deps"
+      "matchPackagePatterns": ["^k8s.io", "^sigs.k8s.io", "^github.com/rabbitmq/cluster-operator", "^github.com/operator-framework/api"],
+      "enabled": false
     }
-  ]
+  ],
+  "postUpgradeTasks": {
+    "commands": ["go mod tidy", "make manifests"],
```
Comment (bogdando) [inline renovate.json:29]: plus make generate?

File: renovate.json [hunk 3]
```diff
@@ -2,33 +2,32 @@
...
   "extends": [
-    "config:base"
+    "config:base",
+    "group:all"
   ],
-  "dependencyDashboard": true,
-  "includeForks": true,
   "logFileLevel": "trace",
   "enabledManagers": ["gomod"],
   "postUpdateOptions": ["gomodTidy"],
```
Comment (gibizer) [inline renovate.json:10]: I would add
```
"constraints": {
    "go": "1.18"
  },
```
to keep bumps compatible with or 1.18 limit.

File: renovate.json [hunk 4]
```diff
@@ -2,33 +2,32 @@
   "timezone": "America/New_York",
   "$schema": "https://docs.renovatebot.com/renovate-schema.json",
   "extends": [
-    "config:base"
+    "config:base",
+    "group:all"
```
Comment (gibizer) [inline renovate.json:6]: yeah that `group:all` seems unnecessary. I ended up with the following grouping in nova:

```
"packageRules": [
    {
      "matchPackageNames": ["github.com/openstack-k8s-operators/nova-operator/api"],
      "enabled": false
    },
    {
      "groupName": "openstack-k8s-operators",
      "matchPackagePrefixes": ["github.com/openstack-k8s-operators"],
      "excludePackageNames": ["github.com/openstack-k8s-operators/nova-operator/api"],
      "schedule": [
        "every weekend"
      ]
    },
    {
      "groupName": "k8s.io",
      "matchPackagePrefixes": [
        "k8s.io",
        "sigs.k8s.io"
      ],
      "schedule": [
        "every weekend"
      ],
      "allowedVersions": "< 1.0.0"
    }
  ]
```
* The first rule disables updating nova/api in nova
* The second rule creates a group for all the openstack-k8s-operators dependency. (excluding nova/api again)
* The third rule groups the k8s.io deps and pin them < 1.0.0 (see the versioning rule in their repo readme, it is changed)
This means the rest will be independent bumps but for nova that is only gomega and ginkgo which we do want to handle independently anyhow.

File: renovate.json [hunk 5]
```diff
@@ -2,33 +2,32 @@
...
-      "extends": ["schedule:monthly"],
-      "groupName": "deps"
+      "matchPackagePatterns": ["^k8s.io", "^sigs.k8s.io", "^github.com/rabbitmq/cluster-operator", "^github.com/operator-framework/api"],
+      "enabled": false
     }
-  ]
+  ],
+  "postUpgradeTasks": {
+    "commands": ["go mod tidy", "make manifests"],
+    "fileFilters": ["go.mod", "go.sum", "**/*.go", "**/*.yaml"],
```
Comment (gibizer) [inline renovate.json:30]: why we need this filters?

File: renovate.json (hunk 3 above)
Comment (gibizer) [reply renovate.json:10]: Also I would add `"dependencyDashboard": true,` to get pinned issue collecting all the pending bumps in a single place.

**Review by gibizer — CHANGES_REQUESTED** [review]

File: renovate.json (hunk 3 above)
Comment (dprince) [reply renovate.json:10]: I can re-add the dashboard. Fwiw I removed it but it is still getting created so it doesn't seem to matter: https://github.com/dprince/openstack-operator/issues/15

File: renovate.json (hunk 2 above)
Comment (dprince) [reply renovate.json]: thanks, added

File: renovate.json (hunk 4 above)
Comment (dprince) [reply renovate.json:6]: hmmm, pretty sure I tried without 'group: all' and it left me with lots of PR's for each operator.

File: renovate.json (hunk 5 above)
Comment (gibizer) [reply renovate.json:46]: ahh so it is not a filter to ignore things :)

**Review by gibizer — APPROVED** [review]
looks good. thanks.

**Pull request merged**